pennylane_lightning_gpu/lightning_gpu.py

```diff
@@ -135,29 +199,131 @@ def __init__(
+            """
+            self._gpu_state.setBasisState(index, use_async)
+
+        def _apply_state_vector_GPU(self, state, device_wires, use_async=False):
+            """Initialize the internal state vector in a specified state on GPU.
+            Args:
+            state (array[complex]): normalized input state of length ``2**len(wires)``
+                 or broadcasted state of shape ``(batch_size, 2**len(wires))``
+            device_wires (Wires): wires that get initialized in the state
+            """
+            # translate to wire labels used by device
+            device_wires = self.map_wires(device_wires)
+            dim = 2 ** len(device_wires)
+
+            state = self._asarray(state, dtype=self.C_DTYPE)
+            batch_size = self._get_batch_size(state, (dim,), dim)
+            output_shape = [2] * self.num_wires
+
+            if batch_size is not None:
+                output_shape.insert(0, batch_size)
+
+            if not (state.shape in [(dim,), (batch_size, dim)]):
+                raise ValueError(
+                    "State vector must have shape (2**wires,) or (batch_size, 2**wires)."
+                )
+
+            if not qml.math.is_abstract(state):
+                norm = qml.math.linalg.norm(state, axis=-1, ord=2)
+                if not qml.math.allclose(norm, 1.0, atol=tolerance):
+                    raise ValueError("Sum of amplitudes-squared does not equal one.")
+
+            if len(device_wires) == self.num_wires and Wires(sorted(device_wires)) == device_wires:
+                # Initialize the entire device state with the input state
+                self.syncH2D(self._reshape(state, output_shape))
+                return
+
+            # generate basis states on subset of qubits via the cartesian product
+            basis_states = np.array(list(product([0, 1], repeat=len(device_wires))))
+
+            # get basis states to alter on full set of qubits
+            unravelled_indices = np.zeros((2 ** len(device_wires), self.num_wires), dtype=int)
+            unravelled_indices[:, device_wires] = basis_states
+
+            # get indices for which the state is changed to input state vector elements
+            ravelled_indices = np.ravel_multi_index(unravelled_indices.T, [2] * self.num_wires)
+
+            # set the state vector on GPU with the unravelled_indices and their corresponding values
+            self._gpu_state.setStateVector(ravelled_indices, state, use_async)
+
+        def _apply_basis_state_GPU(self, state, wires):
+            # Initialize the state vector in a specified computational basis state on GPU directly.
+            # Args:
+            #    state (array[int]): computational basis state of shape ``(wires,)``
+            #        consisting of 0s and 1s.
+            #    wires (Wires): wires that the provided computational state should be initialized on
+            # Note: This function does not support broadcasted inputs yet.
```

Comment:
Can this docstring be formatted in the same style as the others using `""" ... """`